test/e2e/tests/settings.spec.js

```diff
@@ -280,4 +311,77 @@ describe('Settings Search', function () {
       },
     );
   });
+
+  it('User can configure show incoming transaction section in security&privacy section', async function () {
+    await withFixtures(
+      {
+        fixtures: new FixtureBuilder().build(),
+        ganacheOptions: defaultGanacheOptions,
+        title: this.test.title,
+      },
+      async ({ driver }) => {
+        await driver.navigate();
+        await unlockWallet(driver);
+
+        await driver.clickElement(
+          '[data-testid="account-options-menu-button"]',
+        );
+        await driver.clickElement({ text: 'Settings', tag: 'div' });
+        await driver.clickElement({ text: 'Security & privacy', tag: 'div' });
+        await driver.waitForSelector('.settings-page__content__modules');
+
+        // For all networks, incoming transactions configuration should be on
+        await driver.waitForSelector(
+          '[data-testid="incoming-transaction-toggle-enable-all"] > .toggle-button--on',
+        );
```

Comment:
If this toggles on all the toggles, then shouldn't we turn it off before testing the toggle ON feature for other toggles otherwise these tests should be incorrect since the toggle is already ON and we are trying to test if the click toggle them ON, whereas it should be off

https://github.com/MetaMask/metamask-extension/assets/39872794/dc10b836-e27d-468d-94e1-579b2600e412

Reply:
hey Nidhi, there will be a follow up ticket to remove `toggle all` function, as it was defined after the dev work started. And by then we can forget about toggle all function and won't have much confusion around 🙏🏻